src/ActionOpen.java

```diff
@@ -0,0 +1,81 @@
+import java.io.File;
+import java.io.FileReader;
+
+import javax.swing.JFileChooser;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JTextPane;
+
+/*
+ * Class that handles the Open Action
+ */
+public class ActionOpen extends Action {
+
+	// Handles opening of the file
```

Comment:
It's great that these Action subclasses encapsulate the GUI behavior and request. However, the ActionOpen class appears to be taking in a lot of parameters and even the SimpleNotePad object (GUI class). There is a lot of coupling in this class. One possible solution is to extend JTextPane to contain a collection of the JMenuItems that should be in the recent file JMenu. That way we don't have to pass in so many parameters. This practice is software localization. It is the process of adapting existing software to accommodate for our software's specific needs.